### PR TITLE
Book 6

### PR DESCRIPTION
**Pull request by MihaelaBizgan**

### CODE REVIEW

File: src/main/java/controller/UserController.java [hunk 1]
```diff
@@ -2,36 +2,45 @@
...
+    public ResponseEntity<String> getUser(@PathVariable long id) {
+        return ResponseEntity.ok("user1");
+    }
 
+    @GetMapping(UserController.API_NAME)
+    public ResponseEntity<String> getAllUsers() {
+        return ResponseEntity.ok("All users!");
     }
 
     @GetMapping("/hello")
```
Comment (GrazielaLia) [inline src/main/java/controller/UserController.java]: Test this method

**Review by GrazielaLia — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 2]
```diff
@@ -47,6 +47,11 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-thymeleaf</artifactId>
+		</dependency>
```
Comment (stefan2212) [inline pom.xml]: We don't need this dependency.

File: src/main/java/com/booking/app/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @PostMapping(UserController.API_NAME)
+    public ResponseEntity<String> createNewUser() {
+        return ResponseEntity.ok("new user");
+    }
+
+    @PutMapping(UserController.API_NAME + "/{id}")
+    public ResponseEntity<String> updateUser(@PathVariable long id){
+        return new ResponseEntity("User " + id + " has been updated!", HttpStatus.ACCEPTED);
```
Comment (stefan2212) [inline src/main/java/com/booking/app/controller/UserController.java]: For variables interpolation it is better to use String.format() than classic "+".
For each append that you make, a new String object is created in the String pool

File: src/main/java/com/booking/app/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @DeleteMapping(UserController.API_NAME + "/{id}")
+    public ResponseEntity<String> deleteUser(@PathVariable long id) {
+        return new ResponseEntity<>("User " + id + " has been deleted!", HttpStatus.OK);
```
Comment (stefan2212) [inline src/main/java/com/booking/app/controller/UserController.java]: Same as the above comentary.

**Review by stefan2212 — CHANGES_REQUESTED** [review]

File: src/main/java/com/booking/app/controller/UserController.java (hunk 3 above)
Comment (MihaelaBizgan) [reply src/main/java/com/booking/app/controller/UserController.java]: Done

**Review by RazvanLoghin — APPROVED** [review]

**Review by RazvanLoghin — APPROVED** [review]

**Review by RazvanLoghin — APPROVED** [review]

**Review by stefan2212 — APPROVED** [review]

**Review by GrazielaLia — APPROVED** [review]

**Pull request merged**